tests/docs_test.go

```diff
@@ -0,0 +1,71 @@
+// +build system
+// +build docs
+
+package tests
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/mccanne/zq/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var doctests = []struct {
+	name string
+	cmd  string
+
+	expected string
+}{
```

Comment:
This is where the action is. It's a flat list, but it would be easy to add one or more levels of nesting in order to follow the structure of the docs. This could then be reflected in the test output for something like:

```
=== RUN   TestDocExamples/search-syntax/value-match
=== RUN   TestDocExamples/search-syntax/quoted-word
=== RUN   TestDocExamples/processors/cut
...
```